front_end/content.py

```diff
@@ -1988,49 +1983,78 @@ def remove_export_paths(self, zip_file_path, tmp_dir_path):
         if os.path.exists(tmp_dir_path):
             shutil.rmtree(tmp_dir_path, ignore_errors=True)
 
-    def rebuild_exercises(self, assignment_title):
-        sql = '''SELECT e.*
-                 FROM exercises e
-                 INNER JOIN assignments a
-                   ON e.course_id = a.course_id AND e.assignment_id = a.assignment_id
-                 WHERE a.title = ?'''
+    def rebuild_exercises(self, assignment_title=None):
+        if assignment_title:
+            sql = '''SELECT e.*
+                     FROM exercises e
+                     INNER JOIN assignments a
+                       ON e.course_id = a.course_id AND e.assignment_id = a.assignment_id
+                     WHERE a.title = ?'''
+            exercises = self.fetchall(sql, (assignment_title, ))
+
+        else:
+            sql = '''SELECT e.*
+                     FROM exercises e'''
+            exercises = self.fetchall(sql)
 
-        for row in self.fetchall(sql, (assignment_title, )):
+        for row in exercises:
             course = row["course_id"]
             assignment = row["assignment_id"]
             exercise = row["exercise_id"]
-            print(f"Rebuilding course {course}, assignment {assignment}, exercise {exercise}")
 
             exercise_basics = self.get_exercise_basics(course, assignment, exercise)
             exercise_details = self.get_exercise_details(course, assignment, exercise)
 
             text_output, image_output, tests = exec_code(self.__settings_dict, exercise_details["answer_code"], exercise_basics, exercise_details)
 
+            exercise_details["tests"] = [{k: v for k, v in d.items() if k != 'text_output' and k != "image_output"} for d in exercise_details["tests"]]
```

Comment:
These keys have to be removed before the dict is squashed with test_outputs or else all outputs will default to ```None```